core/src/test/java/com/datastrato/gravitino/lock/TestLockManager.java

```diff
@@ -0,0 +1,562 @@
+
+    // Pay attention to the lockCleaner thread.
+    lockManager
+        .treeLockRootNode
+        .getAllChildren()
+        .forEach(node -> lockManager.evictStaleNodes(node, lockManager.treeLockRootNode));
+    Assertions.assertTrue(lockManager.totalNodeCount.get() > 1);
+    Assertions.assertTrue(lockManager.totalNodeCount.get() < lockManager.maxTreeNodeInMemory);
+  }
+
+  private TreeLockNode getTreeNode(TreeLockNode root, int depth) {
+    if (depth == 0) {
+      return root;
+    }
+
+    int i = 0;
+    TreeLockNode result = root;
+    while (i++ < depth) {
+      // We know it only has one child;
+      result = result.getAllChildren().get(0);
+    }
+
+    return result;
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {0, 1, 2, 3})
+  void testCreateTreeLock(int level) throws Exception {
+    LockManager lockManager = new LockManager(getConfig());
+    TreeLockNode rootNode = lockManager.treeLockRootNode;
+    TreeLock treeLock = lockManager.createTreeLock(NameIdentifier.of("a", "b", "c", "d"));
+    treeLock.lock(LockType.READ);
+    treeLock.unlock();
+    checkReferenceCount(rootNode);
+
+    // Start to use concurrent threads to create tree lock.
+    TreeLockNode treeLockNode = getTreeNode(rootNode, level);
+    TreeLockNode spyNode = Mockito.spy(treeLockNode);
+    Mockito.doThrow(new RuntimeException("Mock exception"))
+        .when(spyNode)
+        .getOrCreateChild(Mockito.any());
+
+    if (level == 0) {
+      lockManager.treeLockRootNode = spyNode;
+    } else {
+      int parentLevel = level - 1;
+      TreeLockNode parentNode = getTreeNode(rootNode, parentLevel);
+      parentNode.childMap.put(treeLockNode.getName(), spyNode);
+    }
+
+    CompletionService<Integer> service = createCompletionService();
+    int concurrentThreadCount = 1;
+    for (int i = 0; i < concurrentThreadCount; i++) {
+      service.submit(
+          () -> {
+            for (int j = 0; j < 1000; j++) {
+              Assertions.assertThrows(
+                  RuntimeException.class,
+                  () -> lockManager.createTreeLock(NameIdentifier.of("a", "b", "c", "d")));
+            }
+            return 0;
+          });
+    }
+
+    for (int i = 0; i < concurrentThreadCount; i++) {
+      service.take().get();
+    }
+
+    TreeLockNode lockNode = lockManager.treeLockRootNode;
+    checkReferenceCount(lockNode);
+  }
+
+  @Test
+  void testConfigs() {
+    Config config = getConfig();
+    Mockito.when(config.get(TREE_LOCK_MAX_NODE_IN_MEMORY)).thenReturn(20000L);
+    Mockito.when(config.get(TREE_LOCK_MIN_NODE_IN_MEMORY)).thenReturn(2000L);
+    Mockito.when(config.get(TREE_LOCK_CLEAN_INTERVAL)).thenReturn(2000L);
+
+    LockManager manager = new LockManager(config);
+    Assertions.assertEquals(20000L, manager.maxTreeNodeInMemory);
+    Assertions.assertEquals(2000L, manager.minTreeNodeInMemory);
+    Assertions.assertEquals(2000L, manager.cleanTreeNodeIntervalInSecs);
+  }
+}
```

Comment:
Add a test case for reaching the TREE_LOCK_MAX_NODE_IN_MEMORY limitation.

Reply:
added